### PR TITLE
Implement navigation and tracking for Campaigns dashboard card

### PR DESCRIPTION
**Pull request by AjeshRPai**
Part of 
#18521 
#18525 

## Description 
This PR implements the tracking and click logic in the Campaigns Card

This PR implements the following click events and the tracking associated with it in Campaigns card
- Navigating to Campaign listing page on card click 
- Navigate to Blaze Flow Creation on Create Campaign Click 
- Navigate to Campaign Details on Campaigns item click

Also Implements the Site Menu Navigation 
- Navigate to Campaign listing page when `blaze_campaigns` is enabled 
- Navigate to Blaze Creation flow when `blaze_campaigns` is disabled

Also note that I have added the Skeleton classes for Campaign listing page and Detail page in this PR. There is no need to test these classes as of now since the logic will change in these classes 

### Hide is working as expected, and my site refresh indicator is shown 
- Go to the Jetpack app with the account not having campaigns 
- Go to dashboard 
- Hide promo card 
- Verify that my site is refreshed and the card is hidden 

### Tracking is working as expected when the promo card shown 
- Go to the Jetpack app with the account not having campaigns 
- Go to dashboard 
- Verify promo card is not shown
-  🔵 Tracked: my_site_dashboard_card_shown, Properties: {"type":"promote_with_blaze","subtype":"no_campaigns"}

### Tracking is working as expected when the campaigns card shown 
- Go to the Jetpack app with an account having campaigns 
- Go to dashboard 
-  🔵 Tracked: my_site_dashboard_card_shown, Properties: {"type":"promote_with_blaze","subtype":"campaigns"}

### Click on the Menu item - Blaze campaign enabled 
- Go to the Jetpack app with an account having campaigns 
- Go to dashboard 
- Click on Menu item - Blaze 
- Verify that the user is navigated to the Campaigns listing page 
- Verify that the tracking event is fired 
- 🔵 Tracked: blaze_entry_point_tapped, Properties: {"source":"menu_item"}
- 🔵 Tracked: blaze_campaign_list_opened, Properties: {"source":"menu_item"}

### Click on the Menu item - Blaze campaign disabled 
- Go to the Jetpack app with an account having campaigns 
- Go to dashboard 
- Click on Menu item - Blaze 
- Verify that the user is navigated to the Campaigns listing page 
- Verify that the tracking event is fired 
- 🔵 Tracked: blaze_entry_point_tapped, Properties: {"source":"menu_item"}
- 🔵 Tracked: blaze_overlay_displayed, Properties: {"source":"menu_item"}

### Click on the Dashboard Card - Blaze Campaign 
- Go to the Jetpack app with an account having campaigns 
- Go to dashboard 
- Click on Create Campaign 
- Verify that the user is navigated to Blaze Flow 
- 🔵 Tracked: blaze_entry_point_tapped, Properties: {"source":"dashboard_card"}
- 🔵 Tracked: blaze_overlay_displayed, Properties: {"source":"dashboard_card"}
- Repeat steps 1-2
- Click on Campaign item 
- Verify event is tracked 🔵 Tracked: blaze_campaign_details_opened, Properties: {"source":"dashboard_card"}
- Click on the dashboard card 
- Verify event tracked - 🔵 Tracked: blaze_campaign_list_opened, Properties: {"source":"dashboard_card"}

#### Ripple effect and Padding, Verify that the BlazeCampaignCard UI is as expected - compare it with the [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/18768) or Figma design 
I had to re-arrange and add some columns in `BlazeCampaignCard`.
This is due to the fact that the click ripple of the view doesn’t fill the padding of the view in which click behavior is added. So adding padding to the column and setting it clickable makes the ripple to only fill the content.  In order to fix that, I moved the padding from the parent column to the child view and set the parent clickable which makes the entire content to have the ripple effect. 

## Merge Instructions 
- [ ] Make sure that https://github.com/wordpress-mobile/WordPress-Android/pull/18768 is merged 
- [ ] Update with trunk if needed 
- [ ] Merge as normal

## Regression Notes
1. Potential unintended areas of impact
- Blaze Card - Promo card and Campaign card is not shown in correct time
- Blaze Card UI is not correct 
- Click and tracking for the card is not correct 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Unit tests and Manual tests 

3. What automated tests I added (or what prevented me from doing so)
- Added unit tests

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt [hunk 1]
```diff
@@ -312,9 +312,7 @@ class MySiteViewModel @Inject constructor(
     val onShowJetpackIndividualPluginOverlay = _onShowJetpackIndividualPluginOverlay as LiveData<Event<Unit>>
     val onTrackWithTabSource = _onTrackWithTabSource as LiveData<Event<MySiteTrackWithTabSource>>
     val selectTab: LiveData<Event<TabNavigation>> = _selectTab
-    val refresh = blazeCardViewModelSlice.refresh.observeForever {
-        refresh()
-    }
+    val refresh = blazeCardViewModelSlice.refresh
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: @AjeshRPai - I'd consider renaming this to something to represent blaze

File: WordPress/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1071,6 +1071,12 @@
             android:screenOrientation="portrait"
             android:theme="@style/WordPress.NoActionBar"/>
 
+        <activity android:name=".ui.blazeCampaigns.BlazeCampaignParentActivity"
+            android:exported="false"
+            android:label="@string/blaze_campaigns_page_title"
+            android:screenOrientation="portrait"
```
Comment (zwarm) [inline WordPress/src/main/AndroidManifest.xml:1077]: 💛  We can start with portrait, but we may want to consider landscape support for campaigns.

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityNavigator.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.wordpress.android.ui.blazeCampaigns.ARG_EXTRA_BLAZE_CAMPAIGN_PAGE
+import org.wordpress.android.ui.blazeCampaigns.BlazeCampaignPage
+import org.wordpress.android.ui.blazeCampaigns.BlazeCampaignParentActivity
+import org.wordpress.android.ui.blazeCampaigns.campaigndetail.CampaignDetailPageSource
+import org.wordpress.android.ui.blazeCampaigns.campaignlisting.CampaignListingPageSource
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ActivityNavigator @Inject constructor() {
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityNavigator.kt:14]: ❔ Do you anticipate us adding other activities here or will this be exclusive to Blaze campaigns? If the latter, then we should consider renaming to make it explicit to blaze

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityNavigator.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            Intent(context, BlazeCampaignParentActivity::class.java).apply {
+                putExtra(
+                    ARG_EXTRA_BLAZE_CAMPAIGN_PAGE,
+                    BlazeCampaignPage.CampaignListingPage(campaignListingPageSource)
+                )
+            }
+        )
+    }
+
+    fun navigateToCampaignDetailPage(context: Context, campaignDetailPageSource: CampaignDetailPageSource) {
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityNavigator.kt:26]: 👍 Eventually the campaignId will be passed along here as well.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/BlazeCardViewModelSlice.kt [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import org.wordpress.android.ui.blazeCampaigns.campaigndetail.CampaignDetailPageSource
+import org.wordpress.android.ui.blazeCampaigns.campaignlisting.CampaignListingPageSource
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.BlazeCardBuilderParams
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.BlazeCardBuilderParams.CampaignWithBlazeCardBuilderParams
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.BlazeCardBuilderParams.PromoteWithBlazeCardBuilderParams
+import org.wordpress.android.ui.mysite.MySiteUiState.PartialState.BlazeCardUpdate
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+
+class BlazeCardViewModelSlice @Inject constructor(
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/BlazeCardViewModelSlice.kt:16]: ❤️ ❤️ Love this refactor. Thank you for taking on the challenge

**Review by zwarm — APPROVED** [review]
@AjeshRPai - things are looking good. A few small refactor areas. My compose knowledge is very low, so it's hard for me to say the code is efficient, but it works. :)

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: Done in [fb38a23](https://github.com/wordpress-mobile/WordPress-Android/pull/18783/commits/fb38a23c23f7c2e4e6762f89a04dbe2b4ea5d982)

File: WordPress/src/main/AndroidManifest.xml (hunk 2 above)
Comment (AjeshRPai) [reply WordPress/src/main/AndroidManifest.xml:1077]: 👍🏼  Removed `portrait` in [93788e0](https://github.com/wordpress-mobile/WordPress-Android/pull/18783/commits/93788e00c3992d93d02ddfcc3d6a3a398d971cd4)

**Pull request merged**